developer_guides/firmware/component-tutorial/tut-i-basic-fw-code.rst

```diff
@@ -320,18 +305,30 @@ Signal Processing Function ``amp_copy``
 This first version of the processing function simply copies input samples to
 output and shows how to:
 
-* Use ``struct comp_copy_limits`` to retrieve information about processed
-  frames.
+* Use :cpp:struct:`comp_copy_limits` and :cpp:func:`comp_get_copy_limits_with_lock()`
+  to retrieve information about the number of samples to be processed.
+
+* Refresh the local data cache with :cpp:func:`buffer_invalidate()` in case the
+  input data is being provided to the source buffer by a component running on
+  another core.
 
-* Iterate over the frames, channels, and samples.
+* Iterate over the frames, channels, and samples using :cpp:struct:`comp_copy_limits`
+  descriptor.
```

Comment:
For me `:cpp:struct:` crashes all sphinx-build versions between 2.0.0. and 2.3.1. Which version were you running locally?

Full details and (tiny) workaround in PR #225 , please review.